spec/controllers/articles_controller_spec.rb

```diff
@@ -0,0 +1,234 @@
+
+    before { get :index }
+
+    it 'returns http success' do
+      expect(response).to have_http_status(:success)
+    end
+
+    it 'assigns articles' do
+      expect(assigns(:articles)).to eq(articles)
+    end
+
+    it "renders 'index' template" do
+      expect(response).to render_template(:index)
+    end
+  end
+
+  describe 'GET #show' do
+    let(:article) { create(:article) }
+
+    before { get :show, params: { id: article.id } }
+
+    it 'returns http success' do
+      expect(response).to have_http_status(:success)
+    end
+
+    it 'assigns article' do
+      expect(assigns(:article)).to eq(article)
+    end
+
+    it "renders 'show' template" do
+      expect(response).to render_template(:show)
+    end
+  end
+
+  describe 'POST #create' do
+    let(:params) do
+      {
+        article: {
+          title: title,
+          body: "some example body",
+          status: "public"
+        }
+      }
+    end
+
+    let(:title) { "article title" }
+
+    context 'when valid credentials' do
+      let(:article_attributes) do
+        assigns(:article).attributes.deep_symbolize_keys.except(:id, :created_at, :updated_at)
+      end
+
+      before do
+        request.headers['Authorization'] = valid_credentials
+        post :create, params: params
+      end
+
+      shared_examples 'article assigner' do
+        it 'assigns article' do
+          expect(article_attributes).to eq(params[:article])
+        end
+      end
+
+      context 'when success' do
+        it 'returns http found' do
+          expect(response).to have_http_status(:found)
+        end
+
+        it 'redirects to booking requests' do
+          expect(response).to redirect_to(article_path(id: assigns(:article).id))
+        end
+
+        it_behaves_like 'article assigner'
+      end
+
+      context 'when errors' do
+        let(:title) { "" }
+
+        it 'returns http unprocessable_entity' do
+          expect(response).to have_http_status(:unprocessable_entity)
+        end
+
+        it 'renders new template' do
+          expect(response).to render_template(:new)
+        end
+
+        it_behaves_like 'article assigner'
+      end
+    end
+
+    context 'when invalid credentials' do
+      it_behaves_like 'credentials checker' do
+        before { post :create, params: params }
+      end
+    end
+  end
+
+  describe 'PUT #update' do
+    let(:params) do
+      {
+        id: article.id,
+        article: {
+          title: title,
+          body: "some example body",
+          status: "public"
+        }
+      }
+    end
+
+    let(:title) { "article title" }
+    let(:article) { create(:article) }
+
+    context 'when valid credentials' do
+      before do
+        request.headers['Authorization'] = valid_credentials
+        put :update, params: params
+      end
+
+      shared_examples 'article updater' do
+        it 'assigns article' do
+          expect(assigns(:article)).to eq(article.reload)
+        end
```

Comment:
There is a sense to check `article.reload` attributes so they are eq to sent params.